query/src/main/scala/filodb/query/exec/rangefn/RateFunctions.scala

```diff
@@ -173,6 +173,10 @@ abstract class ChunkedRateFunctionBase extends CounterChunkedRangeFunction[Trans
                     startRowNum: Int, endRowNum: Int,
                     startTime: Long, endTime: Long): Unit = {
     val dblReader = reader.asDoubleReader
+    // if the chunk is a single row NaN value, then return. Prometheus end of time series marker.
+    // This is to make sure we don't set highestValue to zero. avoids negative rate/increase values.
+    if (startRowNum == 0 && endRowNum == 0 && dblReader.apply(acc, vector, startRowNum).isNaN)
```

Comment:
Just curious if check should be
 `if (startRowNum == endRowNum && dblReader.apply(acc, vector, startRowNum).isNaN)`

Reply:
for this issue to manifest, it has to be single row NaN chunk. We could just test for `endRowNum == 0` this implies startRowNum is 0.

Reply:
Thanks @sherali42 . So basically, we are not updating the highestValue in this condition ? 
Should we also add a unit test for this edge case ?

Reply:
@sandeep6189 added a test case, PTAL